src/main/scala/ch/epfl/bluebrain/nexus/kg/storage/S3StorageOperations.scala

```diff
@@ -134,4 +123,47 @@ object S3StorageOperations {
     }
   }
 
+  final class Link[F[_]](storage: S3Storage)(implicit F: Effect[F], as: ActorSystem, config: StorageConfig)
+      extends LinkFile[F] {
+
+    private implicit val ec: ExecutionContext = as.dispatcher
+    private implicit val mt: Materializer     = ActorMaterializer()
+
+    override def apply(id: ResId, fileDesc: FileDescription, location: Uri): F[FileAttributes] =
+      getKey(storage.bucket, location.path) match {
```

Comment:
I guess here we should have a check that the `location` contains the storage endpoint